src/cavendish_particle_tracks/_set_fiducial_dialog.py

```diff
@@ -0,0 +1,536 @@
+        self.lbl_point_depth.setText(str(self.point_depth))
+
+        # region Magnification ##################################
+        self.points[0][0].name = self.cmb_front1.currentText()
+        self.points[0][1].name = self.cmb_front2.currentText()
+        self.a, self.b = magnification(
+            self.points[0][0],  # F1 View 1
+            self.points[1][0],  # F2 View 1
+            self.points[3][0],  # B1 View 1
+            self.points[4][0],  # B2 View 1
+        )
+        # Update the magnification table
+        self.table.setItem(0, 0, QTableWidgetItem(str(self.a)))
+        self.table.setItem(0, 1, QTableWidgetItem(str(self.b)))
+
+    def _on_click_save_to_table(self) -> None:
+        """When 'Save to table' button is clicked, propagate stereoshift and depth to main table"""
+        # Propagate to particle
+        selected_row = self.parent._get_selected_row()
+        self.parent.data[selected_row].spoints = self.spoints
+        self.parent.data[selected_row].shift_fiducial = self.shift_fiducial
+        self.parent.data[selected_row].shift_point = self.shift_point
+        self.parent.data[selected_row].stereoshift = self.point_stereoshift
+        self.parent.data[selected_row].depth_cm = self.point_depth
+
+        # Propagate to parent table
+        for i in range(2):
+            self.parent.table.setItem(
+                selected_row,
+                self.parent._get_table_column_index("sf" + str(i + 1)),
+                QTableWidgetItem(str(self.spoints[i])),
+            )
+            self.parent.table.setItem(
+                selected_row,
+                self.parent._get_table_column_index("sp" + str(i + 1)),
+                QTableWidgetItem(str(self.spoints[i + 2])),
+            )
+        self.parent.table.setItem(
+            selected_row,
+            self.parent._get_table_column_index("shift_fiducial"),
+            QTableWidgetItem(str(self.shift_fiducial)),
+        )
+        self.parent.table.setItem(
+            selected_row,
+            self.parent._get_table_column_index("shift_point"),
+            QTableWidgetItem(str(self.shift_point)),
+        )
+        self.parent.table.setItem(
+            selected_row,
+            self.parent._get_table_column_index("stereoshift"),
+            QTableWidgetItem(str(self.point_stereoshift)),
+        )
+        self.parent.table.setItem(
+            selected_row,
+            self.parent._get_table_column_index("depth_cm"),
+            QTableWidgetItem(str(self.point_depth)),
+        )
+
+    def cancel(self) -> None:
+        """On cancel remove the points_Stereoshift layer"""
+        # TODO: this is a problem, the layer still exists... not sure how to remove it
+        # TODO: Rework this with the new layer handling.
+        self.parent.viewer.layers.select_previous()
+        self.parent.viewer.layers.remove(self.layer_points)
+        return super().accept()
```

Comment:
This is one of the behaviours we want to change, as it would be useful to retain the layer with the current positions of the points for the next time the dialog is called.